accounting_reporting_core/src/main/java/org/cardanofoundation/lob/app/accounting_reporting_core/resource/presentation_layer_service/ExtractionItemService.java

```diff
@@ -39,13 +40,27 @@ public ExtractionTransactionView findTransactionItems(LocalDate dateFrom, LocalD
         );
     }
 
+    @Transactional(readOnly = true)
+    public ExtractionTransactionView findTransactionItemsPublic(String orgId , LocalDate dateFrom, LocalDate dateTo, List<String> event, String currency, BigDecimal minAmount, BigDecimal maxAmount, List<String> transactionHash) {
+
+        List<ExtractionTransactionItemView> transactionItem = transactionItemRepositoryImpl.findByItemAccountDate(orgId ,dateFrom, dateTo,event, currency, minAmount, maxAmount, transactionHash).stream().map(item -> {
+            return extractionTransactionItemViewBuilder(item);
+        }).collect(Collectors.toList());
+
+        return new ExtractionTransactionView(
+                transactionItem.stream().count(),
+                transactionItem
+        );
+    }
+
     private ExtractionTransactionItemView extractionTransactionItemViewBuilder(TransactionItemEntity item) {
         return new ExtractionTransactionItemView(
                 item.getId(),
                 item.getTransaction().getTransactionInternalNumber(),
                 item.getTransaction().getId(),
                 item.getTransaction().getEntryDate(),
                 item.getTransaction().getTransactionType(),
+                item.getTransaction().getLedgerDispatchReceipt().getPrimaryBlockchainHash(),
```

Comment:
I suggest also to return blockchain type but it can of course added later as well